wfe/web-front-end.go

```diff
@@ -914,6 +1098,30 @@ func (wfe *WebFrontEndImpl) BuildID(response http.ResponseWriter, request *http.
 	response.WriteHeader(http.StatusOK)
 	detailsString := fmt.Sprintf("Boulder=(%s %s) Golang=(%s) BuildHost=(%s)", core.GetBuildID(), core.GetBuildTime(), runtime.Version(), core.GetBuildHost())
 	if _, err := fmt.Fprintln(response, detailsString); err != nil {
+		logEvent.Error = err.Error()
 		wfe.log.Warning(fmt.Sprintf("Could not write response: %s", err))
 	}
 }
+
+func (wfe *WebFrontEndImpl) logTerminatedRequest(logEvent requestEvent) {
+	now := time.Now()
+	logEvent.ResponseTime = &now
+	wfe.log.InfoObject(fmt.Sprintf("Terminated request"), logEvent)
+}
+
+func (wfe *WebFrontEndImpl) logIncomingRequest(request *http.Request) (logEvent requestEvent) {
+	now := time.Now()
+	logEvent = requestEvent{
+		ID:           core.NewToken(),
+		RealIP:       request.Header.Get("X-Real-IP"),
+		ForwardedFor: request.Header.Get("X-Forwarded-For"),
+		Method:       request.Method,
+		Extra:        make(map[string]interface{}, 0),
+	}
+	if request.URL != nil {
+		logEvent.Endpoint = request.URL.String()
+	}
+	wfe.log.InfoObject(fmt.Sprintf("Incoming request"), logEvent)
```

Comment:
Logging here doesn't get the Extra object fields. Maybe we should just log the InfoObject in the defer method, and rename this to `populateLogEvent`?